GeometryDash.Server.Serialization.Generator/SerializerGenerator.cs

```diff
@@ -42,11 +42,19 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
         var keyed = ctx.TargetSymbol.GetAttributes()
             .SingleOrDefault(a => a.AttributeClass!.ToDisplayString() == KnownTypes.KeyedAttribute) is not null;
 
+        var symbol = (INamedTypeSymbol)ctx.TargetSymbol;
+        string? baseClassFqn = null;
+        var baseType = symbol.BaseType;
+        if (baseType is not null && baseType.GetAttributes().Any(a => a.AttributeClass?.ToDisplayString() == KnownTypes.SeparatorAttribute))
+        {
+            baseClassFqn = baseType.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat);
+        }
+
```

Comment:
no need for the check (effectively FAWMN does it).

```suggestion
        var classSymbol = (INamedTypeSymbol)ctx.TargetSymbol;
        var baseClassFqn = classSymbol.BaseType?.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat);

```